Jenkinsfile

```diff
@@ -356,6 +354,41 @@ pipeline {
               }
             }
           }
+          stage('Release') {
+            options {
+              skipDefaultCheckout()
+              timeout(time: 12, unit: 'HOURS')
+            }
+            when {
+              beforeInput true
+              when {
+                // Tagged release events ONLY
+                tag pattern: '\\d+\\.\\d+\\.\\d+(-alpha|-beta|-rc)?', comparator: 'REGEXP'
+              }
+            }
+            stages {
+              stage('Notify') {
+                steps {
+                  emailext subject: '[apm-agent-dotnet] Release ready to be pushed',
+                           to: "${NOTIFY_TO}",
+                           body: "Please go to ${env.BUILD_URL}input to approve or reject within 12 hours."
+                }
+              }
+              stage('Release to NuGet') {
+                input {
+                  message 'Should we release a new version?'
```

Comment:
Let's start with the simple use case when the versioning is created beforehand as this is a tagged release event-based.

Reply:
Nit: could we add the version to the message? I.e.: `Should we release "v1.0.0"?` 🤔 

Whenever seeing this kind of screens (do you like to remove the item?), as a user I like having more information in the screen.

Reply:
If we do something like that, it might be required to grab the version from the properties file, `./src/Directory.Build.props` and for such, I see two concerns:
- If that file change somehow in the future, it will require to change the CI pipeline accordingly. I'm not a huge fan of duplicating that.
- It might require some overhead with environment variables to be created on the fly at the very beginning to be able to change the message with that. So a bit hard to maintain it.

On the other hand, the input approval will always refer to the tag release, therefore the CI job name will always contain the version. And also the email notification with the link to the input approval, see https://github.com/elastic/apm-agent-dotnet/pull/696/files#diff-58231b16fdee45a03a4ee3cf94a9f2c3R373 that refers to the `BUILD_URL` env variable.

I'm not saying we should not do it, what do you think if we move forward and then review this release automation process after a few releases?